basket/news/tasks.py

```diff
@@ -329,34 +317,20 @@ def fxa_direct_update_contact(fxa_id, data):
     Ignore if contact with FxA ID can't be found
     """
     try:
-        sfdc.contact.update(f"FxA_Id__c/{fxa_id}", data)
-    except sfapi.SalesforceMalformedRequest as e:
-        if e.content[0]["errorCode"] == "REQUIRED_FIELD_MISSING":
-            # couldn't find the fxa_id and tried to create a record but doesn't
-            # have the required data to do so. We can drop this one.
-            return
-        else:
-            # otherwise it's something else and we should potentially retry
-            raise
-    except SFDCDisabled:
-        pass
-
-    basket_data = from_sfdc(data)
-    try:
-        ctms.update_by_alt_id("fxa_id", fxa_id, basket_data)
+        ctms.update_by_alt_id("fxa_id", fxa_id, data)
     except CTMSNotFoundByAltIDError:
         # No associated record found, skip this update.
         pass
 
 
 @et_task
 def fxa_delete(data):
-    fxa_direct_update_contact(data["uid"], {"FxA_Account_Deleted__c": True})
+    fxa_direct_update_contact(data["uid"], {"fxa_deleted": True, "newsletters": []})
```

Comment:
This doesn't unsubscribe people from newsletters right?

Reply:
Definitely a good call out here. The changes here are attempting to reproduce what the `from_sfdc` code did with the `data` passed in. The tests don't cover any newsletter mapping via this call, but I will dig deeper here.

Reply:
To remove the subscription to all newsletters requires a special string, sending the empty list is basically a no-op. See: https://github.com/mozilla-it/ctms-api/blob/main/tests/unit/test_api_patch.py#L440